src/koheesio/spark/writers/file_writer.py

```diff
@@ -0,0 +1,165 @@
+"""
+File writers for different formats:
+- CSV
+- Parquet
+- Avro
+- JSON
+- ORC
+- Text
+
+The FileWriter class is a configurable Writer that allows writing to different file formats providing any option needed.
+CsvFileWriter, ParquetFileWriter, AvroFileWriter, JsonFileWriter, OrcFileWriter, and TextFileWriter are convenience
+classes that just set the `format` field to the corresponding file format.
+
+"""
+
+from typing import Union
+from enum import Enum
+from pathlib import Path
+
+from koheesio.models import ExtraParamsMixin, Field, field_validator
+from koheesio.spark.writers import BatchOutputMode, Writer
+
+
+class FileFormat(str, Enum):
+    """Supported file formats for the FileWriter class"""
+
+    csv = "csv"
+    parquet = "parquet"
+    avro = "avro"
+    json = "json"
+    orc = "orc"
+    text = "text"
+
+
+class FileWriter(Writer, ExtraParamsMixin):
+    """
+    A configurable Writer that allows writing to different file formats providing any option needed.
+
+    Extra parameters can be passed to the writer as keyword arguments.
+
+    Example:
```

Comment:
For formatting purposes, change this to a header:
~~~markdown
Examples
--------
...
~~~